frontend/src/lib/components/PropertyFilters/propertyFilterLogic.js

```diff
@@ -0,0 +1,84 @@
+import { kea } from 'kea'
+import api from '../../../lib/api'
+import { userLogic } from 'scenes/userLogic'
+
+export const propertyFilterLogic = kea({
+    key: props => props.onChange,
+    connect: {
+        values: [userLogic, ['eventProperties']],
+    },
+    actions: () => ({
+        loadEventProperties: true,
+        setProperties: properties => ({ properties }),
+        update: filters => ({ filters }),
+        setInitial: filters => ({ filters }),
+        set: (index, key, value) => ({ index, key, value }),
+        newFilter: true,
+        remove: index => ({ index }),
+    }),
+
+    loaders: ({ values }) => ({
+        properties: {
+            loadPeopleProperties: async () => {
+                return await api.get('api/person/properties')
+            },
+        },
+    }),
+
+    reducers: ({ actions, values }) => ({
+        properties: [
+            [],
+            {
+                [actions.loadPeoplePropertiesSuccess]: (_, { properties }) =>
+                    properties.map(property => ({
+                        label: property.name,
+                        value: property.name,
+                    })),
+                [actions.setProperties]: (_, { properties }) => properties,
+            },
+        ],
+        filters: [
+            [],
+            {
+                [actions.setInitial]: (_, { filters }) => {
+                    if (!filters) return []
+                    return Object.entries(filters).map(([key, value]) => {
+                        let dict = {}
+                        dict[key] = value
+                        return dict
+                    })
+                },
+                [actions.set]: (filters, { index, key, value }) => {
+                    const newFilters = [...filters]
+                    newFilters[index] = { [key]: value }
+                    return newFilters
```

Comment:
If you want an oneliner for this:
`return filters.map((filter, i) => i === index ? { [key]: value } : filter)`
... but this assumes `index` will always be between `0` and `filters.length() - 1`, which might be the expected case

Reply:
I do love a good one-liner but I'm not sure it's more readable.

Reply:
Once you use `map` a lot, it'll look quite elegant 🤪
Up to you though, the existing code is fine.